syntax = "proto3";

package onetake.reco.v1;

option csharp_namespace = "OneTake.GrpcContracts.Reco.V1";

service RecoService {
  rpc GetRecommendations(GetRecommendationsRequest) returns (GetRecommendationsResponse);
}

message GetRecommendationsRequest {
  string user_id = 1;
  int32 limit = 2;
  string feed_type = 3;   // e.g. HOME | FOLLOWING | POST_DETAILS
  string context_post_id = 4;  // optional
  repeated string context_tags = 5;  // optional
  repeated string exclude_post_ids = 6;  // optional
  string trace_id = 7;
}

message RecommendationItem {
  string post_id = 1;
  double score = 2;
  string reason = 3;
}

message GetRecommendationsResponse {
  repeated RecommendationItem items = 1;
}
